pkg/password/password_policies.go

```diff
@@ -0,0 +1,167 @@
+package password
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"unicode/utf8"
+)
+
+var defaultSpecialCharacters = "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+
+type Validator interface {
+	Validate(str string) error
+}
+
+type Policies struct {
+	minCharacters            int
+	minLowerCaseCharacters   int
+	minUpperCaseCharacters   int
+	minDigits                int
+	minSpecialCharacters     int
+	allowedSpecialCharacters string
+	digits                   *regexp.Regexp
+	specialCharacters        *regexp.Regexp
+}
+
+func NewPasswordPolicies(minCharacters, minLowerCaseCharacters, minUpperCaseCharacters, minDigits, minSpecialCharacters int,
+	allowedSpecialCharacters string) (Validator, error) {
```

Comment:
Remove `allowedSpecialCharacters` and always use the default. I don't see a usecase for custom special characters.

Reply:
I discussed with Tobias and Michael. So we have to keep it and the escaping rules have to be documented.

Reply:
Really? Can't we try to convince them that this is a bad idea? Nobody will use it and the ones that use it will run into issues...